### PR TITLE
Add array examples to Storm lift / filter docs

### PR DESCRIPTION
**Pull request by therealsilence**
- Add array "by" syntax section / examples to Storm lift and filter docs
- Slight updates / changes to "array" section of Storm type-specific behavior docs
- Tidy jupyter cells in lift and filter docs

### CODE REVIEW

File: docs/synapse/userguides/storm_ref_lift.ipynb [hunk 1]
```diff
@@ -20,6 +21,19 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "hideCode": true,
+    "hideOutput": true
+   },
+   "outputs": [],
+   "source": [
+    "#Create a cortex which should contain the runt nodes for the data model\n",
```
Comment (Cisphyx) [inline docs/synapse/userguides/storm_ref_lift.ipynb]: Missing a space after the #

File: docs/synapse/userguides/storm_ref_lift.ipynb [hunk 2]
```diff
@@ -72,9 +86,9 @@
    },
    "outputs": [],
    "source": [
-    "core = await getTempCoreCmdr()\n",
+    "#make some nodes\n",
```
Comment (Cisphyx) [inline docs/synapse/userguides/storm_ref_lift.ipynb]: Formatting is different than the rest of the "# Make some nodes:\n" comments

File: docs/synapse/userguides/storm_ref_type_specific.ipynb [hunk 3]
```diff
@@ -335,7 +340,7 @@
    "cell_type": "raw",
    "metadata": {},
    "source": [
-    "Pivot from the domain(s) referenced by a set of x509 certificates the node for those domains:"
+    "Pivot from the domain(s) referenced by a set of x509 certificates the nodes for those domains:"
```
Comment (Cisphyx) [inline docs/synapse/userguides/storm_ref_type_specific.ipynb]: I think there is a 'to' missing here

File: docs/synapse/userguides/storm_ref_type_specific.ipynb [hunk 4]
```diff
@@ -918,9 +923,10 @@
...
-    "- Domains can be lifted or filtered using the regular expression (regex) extended operator (``~=``), though lifting in particular may impose a performance overhead (see :ref:`lift-regex` and :ref:`filter-regex`).\n",
-    "- Suffix matching is not supported for filter operations (note that a \"lift by suffix\" is effectively a combined \"lift and filter\" operation)."
+    "- Because domains are reverse-indexed instead of prefix indexed, for **lift** operations, partial string matching can only occur based on the end (suffix) of a domain. It is not possible to lift by a string at the beginning of a domain. For example, ``inet:fqdn=\"yahoo*\"`` and ``inet:fqdn^=yahoo`` are both invalid.\n",
+    "- Domains can be **filtered** by prefix (``^=``). For example, ``inet:fqdn=\"*.biz\" +inet:fqdn^=smtp`` is valid.\n",
+    "- Doomains cannot be filtered based on suffix matching (note that a \"lift by suffix\" is effectively a combined \"lift and filter\" operation).\n",
```
Comment (Cisphyx) [inline docs/synapse/userguides/storm_ref_type_specific.ipynb]: Extra 'o' in Doomains

Comment (vEpiphyte) [reply docs/synapse/userguides/storm_ref_type_specific.ipynb]: Those are the Domains of Mt. Doom

Comment (therealsilence) [reply docs/synapse/userguides/storm_ref_type_specific.ipynb]: "doomains" = malicious domains.

j/k fixed

**Pull request merged**